spec/requests/restaurants_spec.rb

```diff
@@ -0,0 +1,132 @@
+  end
+
+  describe 'GET /restaurants' do
+    before do
+      get restaurants_path
+    end
+
+    it 'return https success' do
+      expect(response).to have_http_status(200)
+    end
+
+    it 'has the right objects count' do
+      arr = response_body
+      expect(arr.count).to eq(Restaurant.all.count)
+    end
+
+    it 'contains mandatory attributes' do
+      arr = response_body
+      arr.each do |json_rest|
+        expect(json_rest.keys).to include('id', 'name', 'cuisine', 'accept10bis')
+      end
+    end
+  end
+
+  describe 'GET /restaurant/:id' do
+    subject { Restaurant.all.sample }
+    it 'return https success' do
+      get restaurant_path(subject.id)
+      expect(response).to have_http_status(200)
+    end
+
+    it 'returns :not_found status on invalid uuid' do
+      get restaurant_path('this-is-not-a-uuid')
+      expect(response).to have_http_status(:not_found)
+    end
+
+    it 'returns the correct record' do
+      get restaurant_path(subject.id)
+      restaurant_matcher(subject, response_body)
+    end
+  end
+
+  describe 'POST /restaurants' do
+    it 'returns http success' do
+      post '/restaurants', params: FactoryBot.attributes_for(:restaurant)
+      expect(response).to have_http_status(:created)
+    end
+
+    it 'creates new record in the database' do
+      post '/restaurants', params: FactoryBot.attributes_for(:restaurant)
+      uuid = response_body['id']
+      rest = Restaurant.find_by(id: uuid)
+      expect(rest).to_not be_nil
+      restaurant_matcher(rest, response_body)
+    end
+
+    it 'returns error when params are invalid' do
+      invalid_attr = FactoryBot.attributes_for(:restaurant, :invalid_record)
+      post '/restaurants', params: invalid_attr
+      expect(response).to have_http_status(:unprocessable_entity)
+    end
+  end
+
+  describe 'PUT /restaurant/:id' do
+    subject { Restaurant.all.sample }
+
+    it 'returns https success' do
+      put restaurant_path(subject.id), params: FactoryBot.attributes_for(:restaurant)
+      expect(response).to have_http_status(:no_content)
+    end
+
+    it 'returns error when params are invalid' do
+      invalid_attr = FactoryBot.attributes_for(:restaurant, :invalid_record)
+      put restaurant_path(subject.id), params: invalid_attr
+      expect(response).to have_http_status(:unprocessable_entity)
+    end
+
+    it 'updates the record fields' do
+      new_rest = FactoryBot.build(:restaurant, id: subject.id)
+      put restaurant_path(subject.id), params: new_rest.attributes
+      rest = Restaurant.find(subject.id)
+      restaurant_matcher(rest, new_rest)
+    end
+  end
+
+  describe 'DELETE /restaurant/:id' do
+    subject { Restaurant.all.sample }
+
+    it 'returns http success' do
+      delete restaurant_path(subject.id)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'fails to find the record in DB' do
+      delete restaurant_path(subject.id)
+      expect(Restaurant.find_by(id: subject.id)).to be_nil
```

Comment:
this line does not check what happened when you try to delete a not existing record.
I would change this test to:
1. expect not to find a restaurant with id X
2. try to delete restaurant with id X
3. expect got an error/exception or a not_found status

Reply:
I though about adding this test however it is somehow duplicity of the GET :id test since it will fail on the find(:id).
on the other hand if test scopes shouldn't deal with implementation phases considerations this test is required...
anyways... adding it   -_-

Reply:
regarding the duplicity - it's a unit test so we are testing a specific logic (in this test `delete`) and we are assuming that everything else works perfectly (and of course we have tests for the function we use)

let me be more clear, `find(:id)` only uses us to make sure that we are in a "clean" state, and we don't have leftovers from other tests. it's not a must, `delete restaurant_path('nonexisting-uuid')` will work also.